### PR TITLE
Add `KeyboardConfigPointer` typedef and propagate the config to `WTrackMenu`

### PR DESCRIPTION
**Pull request by fwcd**
While looking into customizing the keyboard shortcuts for `WTrackMenu`, I ran into the issue that there was no `ConfigObject<ConfigValueKbd>` in immediate reach. So I started propagating it through the application all the way to the core services. The change became quite sweeping, however, since `WTrackMenu` has quite a few (transitive) dependents.

To avoid relying too much on raw pointers, I've added a `KeyboardConfigPointer` wrapper around `std::shared_ptr<ConfigObject<ConfigValueKbd>>`.

This PR doesn't add any new shortcuts or features since the change is quite large in and of itself and I'd be curious to hear some opinions on whether this is a fine approach before e.g. making the track properties shortcut etc. customizable.

### CODE REVIEW

File: src/library/rhythmbox/rhythmboxfeature.h [hunk 1]
```diff
@@ -18,44 +20,46 @@ class BaseExternalPlaylistModel;
 class RhythmboxFeature : public BaseExternalLibraryFeature {
     Q_OBJECT
  public:
```
Comment (daschuer) [inline src/library/rhythmbox/rhythmboxfeature.h:22]: It looks like this indentation has tricked clang-format. 
Here we use two space indent. 
The rest should have four spaces as before.

**Review by daschuer — CHANGES_REQUESTED** [review]
I have just found a style issue. The rest looks good. 
Thank you for this tedious work.

File: src/widget/wmainmenubar.h [hunk 2]
```diff
@@ -92,7 +92,7 @@ class WMainMenuBar : public QMenuBar {
 
     UserSettingsPointer m_pConfig;
     QAction* m_pViewKeywheel;
-    ConfigObject<ConfigValueKbd>* m_pKbdConfig;
+    const KeyboardConfigPointer m_pKbdConfig;
```
[the remark this answers is not in the document]
Comment (Swiftb0y) [reply src/widget/wmainmenubar.h:95]: For that exact reason, I'm also proposing to deprecate the `*Pointer` convention. It does little more than obstruct the actual semantics of the kind of smart pointer it actually represent as well as the actual type (with cv-qualifications) it actually contains.

[the remark this answers is not in the document]
Comment (fwcd) [reply src/widget/wmainmenubar.h:95]: That's fair, I've mostly tried to follow the existing style `UserSettingsPointer`, also w.r.t. `const`-ness here

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply src/widget/wmainmenubar.h:95]: I appreciate the dedication to consistency, but as I outlined I think that convention was established in error. Possibly from knowledge how cv-qualifiers work on plain pointers vs smart_pointer (or class-templates in general). 
[This Godbolt](https://compiler-explorer.com/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGEgMykrgAyeAyYAHI%2BAEaYxBIA7KQADqgKhE4MHt6%2BASlpGQKh4VEssfFcSXaYDplCBEzEBNk%2BflyB1bUC9Y0ExZExcYm2DU0tue0jvf2l5YkAlLaoXsTI7Bzm/mHI3lgA1Cb%2BbmwsJACeh9gmGgCC1zcKBMReDnvKGAcJVrd7v3thBD2ADcxIdvncEgARMH3e6PZ6vIQsXqfcF/PZoBiPPaPdAgEAKBCNTDoAD6ySehzc73Qlz2FOIMNuJihTLut3hL0BygMYVR93RmOxNIAVPTKf5wSzoZLYczbkDUHh0DjkU0IEiUQp5vyfn8APT6vb0YCgvW/BQAOgZAFpLiDaAd/JC9mY2QKDUa0CxknRMHs4sQSKR6UrBHF/goMQJHiGTuVIwwwBwCB7foacdank6Xbj8ciANaYUmE4m0o40y4QLjzd2s%2BU3RXK%2Bm8hgQHlMPnJHUstGe/60E1iOOYMoRvBRoUEEOpAHjhRJlNpvYZ5JZ4h2/zYB0513u80ro1D2gzsMEed7BioQFTy3LtcM3fhADubw%2Bva40rrkI4i1onAAVl4PwOC0UhUE4NxrGsHFllWf1Nh4UgCE0X9FgLEB/H8S0sNwvD8IANn0ThJGA1DwM4XgFBADRkNQxY4FgJBvV9egyAoCAWL9eJgC4MwzFILAgTwNYADU8EwZ8AHlkkYTgkJoWhz2IaiIGicjojCRoznk3hNOYYgzik6JtBqFDuF4b02EEKSGFoHTQN4LBkSMcRHMEvBiDMvAgUwaj3MwVQai8c9dPIcN/3c2g8GiYhtI8LByKePAWF0xYqAMYAFHEySZLkiyZEEEQxHYKRCvkJQ1HI3QuH0QxjGgyx9Bi6jIEWVAKUyfyqMwexHGcCBXHGNoggYdAZkGCp8nSfqsk8Vo9FSGbMgmscps6WaejGeaJlsXrvO6UY%2BjCAY1r0R5emG86jtWuYayWFY1gkP9ALI9yII4PZVAADgIm0CMkPZgGQZA9j4y0zD2CBcEIEgDjMfwa14cytHmdDMOw/CsdwojItI0gQLAj6qJoujHIYmBEBAZYCGSELyEoLi2IiVh1h%2Bv6AaBkGwbMCHeBJWHiGVPR%2BCK0RxDK0WKpUdR3Jq0hnzi5I0uIjggIJ8iPqkkK6cBVAqC%2B37/sB4HQfByGIA8H1uPhxH5mR%2Bj0awnDsax1X8cJ3hidsUmUbQ1WzDeonKLJ1HFl8lTMhASQgA%3D%3D) demonstrates how I came to that conclusion.

[the remark this answers is not in the document]
Comment (Swiftb0y) [reply src/widget/wmainmenubar.h:95]: (I created a thread on the Zulip to discuss this with the rest of the team https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/const.20.60*Pointer.60.20aliases.20.26.20and.20their.20removal/near/341501181)

**Comment by ronso0** [comment]
Many conflicts have developed while this PR has been stale for a year, I will close it.
@fwcd no offense, please reopen if you're motivated to pick this up again.

**Pull request closed without merging**